CollisionOperator/CollisionProcessors/neutronCEimp_class.f90

```diff
@@ -452,43 +472,73 @@ subroutine cutoffs(self, p, collDat, thisCycle, nextCycle)
     type(collisionData), intent(inout)   :: collDat
     class(particleDungeon),intent(inout) :: thisCycle
     class(particleDungeon),intent(inout) :: nextCycle
+    real(defReal), dimension(3)          :: val
+    real(defReal)                        :: minWgt, maxWgt, avWgt
 
-    if (p % E < self % minE ) then
+    if (p % E < self % minE) then
       p % isDead = .true.
+    ! Weight Windows treatment
+    elseif (self % weightWindows) then
+      val = self % weightWindowsMap % at(p)
+      minWgt = val(1)
+      maxWgt = val(2)
+      avWgt  = val(3)
```

Comment:
What will happen if the user does something stupid with a map and p will be in an invalid bin and all this entries will become 0? 

Is this what the `(maxWgt /= ZERO)` check is for below? 
If so I think it would be helpful to make a comment about this edge case.